infra/test/Test/Pos/Infra/Gen.hs

```diff
@@ -111,3 +134,130 @@ genHandlerSpec = Gen.choice [ ConvHandler <$> genWord16
                                   <$> Gen.word8 (Range.constant 1 255)
                                   <*> gen32Bytes
                             ]
+
+----------------------------------------------------------------------------
+-- Pos.Infra.Network Generators
+----------------------------------------------------------------------------
+
+genDomain :: Gen DNS.Domain
+genDomain = Gen.utf8 (Range.constant 1 127) Gen.alpha
+
+genDnsDomains :: Gen (DnsDomains DNS.Domain)
+genDnsDomains = DnsDomains <$> Gen.list (Range.constant 1 10) singletonNA
+  where
+    singletonNA = Gen.list (Range.singleton 1) (genNodeAddr genDomain)
+
+genMaxBucketSize :: Gen MaxBucketSize
+genMaxBucketSize = Gen.choice
+    [ pure BucketSizeUnlimited
+    , BucketSizeMax <$> Gen.int (Range.constant 1 100)
```

Comment:
This could probably be `linear` so that it scales w/ the Size param. A couple other spots in this module could also benefit from switching, so small tests go faster.